src/tasks/components/TaskRunsRow.tsx

```diff
@@ -85,6 +90,12 @@ class TaskRunsRow extends PureComponent<Props, State> {
     this.setState({isImportOverlayVisible: !this.state.isImportOverlayVisible})
   }
 
+  private handleRetry = async () => {
+    const {retryTask, taskID, run} = this.props
+    await retryTask(taskID, run.id)
+    window.location.reload()
```

Comment:
we should just dispatch to get all the runs rather than reloading the page

Reply:
yeah i saw the other comment, i just need to add that functionality

Reply:
rather than creating a new way of doing things for only this method (and thus making the user experience slightly different), let's just follow convention with tasks and push to history. If we need to change that convention (and I suspect we will) we can change all of those at the same time as a logical change, rather than having yet another disjointed experience.

Reply:
I might be wrong (not at a computer), but isn’t there a thunk to get the runs for a task that we can use to make for a smoother UX?

Reply:
there might be. The point I'm trying to make is, making that change is separate from making this change, which is about retries. There's already enough being changed in this PR in a part of the app we haven't spent a ton of time in without introducing a further fundamental change.

I think it'll be a smoother experience to refresh the data rather than reloading the whole page(I don't think it will be earth-shatteringly better, it'll be a marginal improvement at best), but I don't think it's worth the benefit to confuse the purpose of this PR. We can follow up with a UX change if we need to

Reply:
This sounds like the definition of tech debt to me

Reply:
>My question is, why not just dispatch an event?

Like I said in the comments above, coming up with a different UX flow for this one part of tasks is beyond the scope of this. And because keeping things consistent, even though it **might be** less than ideal, (don't forget that this is just your opinion on what is ideal, this hasn't been discussed with our product manager or designer) is preferable to having code inconsistencies in a part of the app we don't spend a lot of time maintaining and that will most likely be changing in the future.

All the pages under tasks currently do this history-pushing behavior. Why aren't you worried about them? This feature was (supposed to be) a small, quick hit. If we need to iterate on tasks, we should plan to do that separately. If you think this is important, then feel free to prioritize working on [this ticket](https://github.com/influxdata/ui/issues/1388) rather than requesting scope creep on a PR that has been approved.

Reply:
I think I might be misunderstanding you. I don't believe there's another case of the UI using `history.push` to effectively **reload a page**, which is what the original implementation was for this feature. It looks like it has since been updated to use `history.push` to navigate to the task edit page. If that's in fact what the product and design goal of this is (rerun the task and then be redirected to the edit page), then that's not a problem. My problem is that we were previously trying to reload a page in order to refresh the runs for a task. If we are talking about opinions, I'm of the opinion that we shouldn't have our users bearing the cost of engineering by sticking to the conventions of our app when adding in new functionality